ldap_ban.yml

```diff
@@ -1,37 +1,64 @@
 # -*- eval: (ansible) -*-
 - include: vault.yml load=ldap hosts=ldap.hashbang.sh
 
-# This is a sucky way to make LDAP changes,
-# but Ansible's ldap_attr module is v2.3 only
-- hosts: ldap.hashbang.sh
-  become: false
-  vars:
-    ldif: |
-      dn: uid={{ user | mandatory }},ou=People,dc=hashbang,dc=sh
-      changetype: modify
-      replace: loginShell
-      loginShell: /usr/sbin/nologin
+- name: Parse parameters
+  hosts: ldap.hashbang.sh,shell_servers
+  gather_facts: no
 
+  vars_prompt:
+    - name: users
+      prompt: Comma-separated list of users to ban
+      private: no
+
+    - name: delete
+      prompt: Delete home directories? (yes/no)
+      private: no
+      default: "no"
+
+  tasks:
+    - name: Parse users list
+      set_fact:
+        user_list: "{{ users.split(',') }}"
+        delete_homedirs: "{{ delete }}"
+
+- name: Disable the account in LDAP
+  hosts: ldap.hashbang.sh
+  gather_facts: no
   tasks:
-    - name: Disable account in LDAP
-      command: >
-        docker exec -i slapd ldapmodify
-          -D {{ ldap.admin.dn       | quote }}
-          -w {{ ldap.admin.password | quote }}
-      args:
-        stdin: "{{ ldif }}"
+    - ldap_attr:
+        server_uri: ldaps://ldap.hashbang.sh
+        bind_dn: "{{ ldap.admin.dn }}"
+        bind_pw: "{{ ldap.admin.password }}"
+
+        dn: uid={{ item }},ou=People,dc=hashbang,dc=sh
+        name: loginShell
+        state: exact
+        values: /usr/sbin/nologin
+      delegate_to: localhost
+      with_items: "{{ user_list }}"
 
 - hosts: shell_servers
   become: true
+  gather_facts: no
   tasks:
-    - name: Invalidate SSSd cache for {{ user }}
-      command: sss_cache -u {{ user }}
+    - name: Invalidate SSSd cache entries
+      command: sss_cache -u {{ item }}
+      with_items: "{{ user_list }}"
       changed_when: false
+      ignore_errors: yes
 
-    - name: Terminate sessions for {{ user }}
-      command: loginctl terminate-user {{ user }}
+    - name: Terminate user sessions
+      command: loginctl terminate-user {{ item }}
       register: terminate
+      with_items: "{{ user_list }}"
       failed_when: |
         'Could not terminate user: No user' not in terminate.stderr
         and terminate|failed
       changed_when: terminate.stderr == "" and terminate|success
```

Comment:
can we `s/|/ is /` to get rid of the Ansible lint warning?

Reply:
Oh, I didn't notice the lint.  How come it didn't fail the CI?

Reply:
I just noticed I missed the same fix in `changed_when`.